packages/cozy-client/src/models/contact.js

```diff
@@ -247,3 +248,21 @@ export const getIndexByFamilyNameGivenNameEmailCozyUrl = contact => {
  * @returns {boolean}
  */
 export const isContact = doc => doc._type === CONTACTS_DOCTYPE
+
+/**
+ * @param {CozyClient} client - Instance of CozyClient
+ * @param {object} contact - A contact
+ * @returns {Promise<object>} - Modified or deleted contact
+ */
+export const deleteContact = (client, contact) => {
+  const syncData = get(contact, 'cozyMetadata.sync', {})
+  const isLinkedToAccounts = Object.keys(syncData).length > 0
+  if (isLinkedToAccounts) {
+    return client.save({
+      ...contact,
+      trashed: true
+    })
+  } else {
+    return client.destroy(contact)
+  }
+}
```

Comment:
There is already a ContactsCollection, why not adding the destroy method there?

Reply:
Oh well... Looks like I was a bit late https://github.com/cozy/cozy-client/pull/1458/files#r1548113459